packages/flutter/lib/src/widgets/widget_state_mixin.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+import 'widget_state.dart';
+
+/// Mixin for [State] classes that require knowledge of changing [WidgetState]
+/// values for their child widgets.
+///
+/// This mixin does nothing by mere application to a [State] class, but is
+/// helpful when writing `build` methods that include child [InkWell],
+/// [GestureDetector], [MouseRegion], or [Focus] widgets. Instead of manually
+/// creating handlers for each type of user interaction, such [State] classes can
+/// instead provide a `ValueChanged<bool>` function and allow [WidgetStateMixin]
+/// to manage the set of active [WidgetState]s, and the calling of [setState]
+/// as necessary.
+///
+/// See also:
+///
+///  * [MaterialStateMixin], the Material version of `WidgetStatesController`.
+@optionalTypeArgs
+mixin WidgetStateMixin<T extends StatefulWidget> on State<T> {
```

Comment:
This isn't used much anymore. It has been superceded by MaterialStatesController.